### ISSUE TITLE
Update the metrics!

### ISSUE THREAD

**Issue by YLaCoon** [cc-hpc-itwm/DeepFakeDetection#9]
This is a repost as my issues were closed without being resolved. I am not happy with this behavior. It's not scientific.

> As already mentioned, the training data is balanced and this is the important point.

I disagree. The important point is the test performance. The chosen metric "accuracy" is not able measure the performance as the test data is unbalanced. I thus proposed that you switch to `roc_auc_score` as metric! Using it is as simple as `from sklearn.metrics import roc_auc_score`. I look forward to your updates and scores given this meaningful metric for unbalanced data!

**Comment by RicardDurall** [comment]
We thought that with the provided answer the issue was solved. Sorry about that.
If you are still not convinced, you could take the balanced dataset and split it into training and testing.
One possible way would be:

import numpy as np
from sklearn.svm import SVC
import pickle

pkl_file = open('train_3200.pkl', 'rb')
data = pickle.load(pkl_file)
pkl_file.close()
X = data["data"]
y = data["label"]

#test
X_ = np.concatenate((X[0:900],X[2700:3600]),  axis=0)
y_ = np.concatenate((y[0:900],y[2700:3600]),  axis=0)

#train
X = X[900:2700]
y = y[900:2700]


svclassifier_r = SVC(C=6.37, kernel='rbf', gamma=0.86)
svclassifier_r.fit(X, y)

SVM = svclassifier_r.score(X_, y_)
print("SVM: "+str(SVM))

Let us know, if you still have any issue. Otherwise, we will close it.

**Comment by YLaCoon** [comment]
Thank you for addressing this issue more seriously!

Using your split I get a roc_auc_score of 0.947408888888889! That's remarkable and at the same time makes me wonder. In particular, how do you make sure the computed features in X are not from the same video / same person? That would create a severe leakage. The model would be trained and tested on almost the same data.

If you provide the missing information on how the splits and this split in particular were created I will close the issue.

**Comment by RicardDurall** [comment]
If you have a look at FaceForensic.ipynb, you will see that we first create the fake data and then the real. For each video (there is a folder) we take a number of faces that ranges between 1-20 and we save them in our pkl file (train_3200.pkl).  As a result, we end up having from 0:1600 fake and 1600:3200 real features. Now lets say that we want a split of 50% training and 50% testing. We know that we need 800 real and 800 fakes for training and the same for testing. So we do the following splitting

#Test
X_ = np.concatenate((X[0:800],X[2400:3200]),  axis=0)
y_ = np.concatenate((y[0:800],y[2400:3200]),  axis=0)

#Train
X = X[800:2400]
y = y[800:2400]

We can do this easy splitting because we havent shuffled the data.

To be 100% correct, for this example we should use:
X = X[820:2380]
y = y[820:2380]

Note that now we use 20 samples less on training to avoid any kind of overlapping.
I hope that now that everything is now clear!

**Comment by YLaCoon** [comment]
>  I hope that now that everything is now clear!

No, it's **not** possible to verify, sorry. Your recent commits introduced the folders 'prepro_deepFake/real/' and 'prepro_deepFake/fake/'. Please provide a way to create or download them.

**Comment by RicardDurall** [comment]
yesterday I uploaded a link where you can download the folder. Under DeepFakeDetection

**Comment by YLaCoon** [comment]
Removed, see below.

**Comment by keuperj** [comment]
> Thanks for updating the code and providing the missing pre-processing steps! I was suspicious as the provided approach only reached auc of 0.5 for real world data (random guessing). Unfortunately, it looks like your approach only works with synthetic data. In case you want to check, the dataset is attached.
> [youtube.zip](https://github.com/cc-hpc-itwm/DeepFakeDetection/files/4360270/youtube.zip)

That's it - I had enough of your false, unproven accusations! We have been very patient with you and your very unfriendly stated requests. But I will no longer accept that you publicly discredit our work with unsubstantial accusations - this is scientific misconduct! 

I know your sloppy work and just because you can't get our approach to work on your data does not say anything! We have shown that it work on three independent (real) data sets! Working in the frequency domain requires some care, and people who manage to mess up the computation of an FFT should be careful to blame others when they don't get things to work!

You are not  welcome here anymore.

**Comment by YLaCoon** [comment]
I am sorry for the unfriendly tone. I thought that you were just closing issues without addressing them. The recent commits made it easy to reproduce your results. Thank you.
